utils/resolution.go

```diff
@@ -43,3 +43,23 @@ func (r *Resolution) EnsureEven() {
 		r.Width = r.Width + 1
 	}
 }
+
+// ResizeToFit returns the biggest ressolution in the aspect ratio of the source
+// that fits into this resolution, while keeping the aspect ratio the same as the source
+func (r *Resolution) ResizeToFit(source Resolution) Resolution {
+	aspect := float32(source.Width) / float32(source.Height)
+
+	out := Resolution{
+		Width:  r.Width,
+		Height: r.Height,
+		File:   source.File,
```

Comment:
I would actually expect the returned Resolution to be a copy of `r` with only width and height modified.
I.e. file should be r.File